src/include/catalog/database_catalog.h

```diff
@@ -615,6 +616,21 @@ class DatabaseCatalog {
    */
   void BootstrapProcContexts(common::ManagedPointer<transaction::TransactionContext> txn);
 
+  /**
+   * Internal helper method to reduce copy-paste code for populating proc contexts. Allocates the FunctionContext and
+   * inserts the pointer.
+   * @param txn transaction to insert into catalog with
+   * @param proc_oid oid to associate with this proc's and its context
+   * @param func_name Name of function
+   * @param func_ret_type Return type of function
+   * @param args_type Vector of argument types
+   * @param builtin Which builtin this context refers to
+   * @param is_exec_ctx_required true if this function requires an execution context var as its first argument
+   */
+  void BootstrapProcContext(common::ManagedPointer<transaction::TransactionContext> txn, proc_oid_t proc_oid,
+                            std::string &&func_name, type::TypeId func_ret_type, std::vector<type::TypeId> &&args_type,
+                            execution::ast::Builtin builtin, bool is_exec_ctx_required);
```

Comment:
When `is_exec_ctx_required` true? The constructor we rely on for `FunctionContext` has a default arg (separate debate) so it's assigned inconsistently. My suspicion was that string functions would need it to allocate, but that wasn't always the case. Could use a sanity check.

Reply:
Documenting the answer from Slack:

> Joseph Koshakow From what I saw, all string functions were getting an ExecutionContext. Some probably don't need it, but it made Semantic checking slightlier easier in that we group all of the string functions together and for all of them check that their first argument is an execution context.
> Matt Butrovich hm, okay. so worst case scenario we're passing it in and we dont actually need it
> Joseph Koshakow Yeah I think for a lot of them we just pass it through and never use it. It can definitely be cleaned up IMO.  Like exp for example takes the execution context, but then in the bytecode_generator just ignores it.

Reply:
I'm replying here again for recording purposes. Some string functions need it to allocate memory for new strings, and the NpRunners need it as well to access the output buffer. We end up passing it to all string functions for simplicity (which is not necessarily a good approach). Some other functions, like `exp` get it passed by mistake and just ignore it.